File: MessageBus/messagebus-client/src/models/QueueStatus.ts
/**
 * Represents the status of a message queue
 */
export interface QueueStatus {
  /** Name of the queue */
  queueName: string;
  /** Total number of messages in the queue */
  messageCount: number;
  /** Number of messages ready for processing */
  readyCount: number;
  /** Whether receive is enabled on the queue */
  isReceiveEnabled: boolean;
  /** Whether activation is enabled on the queue */
  isActivationEnabled: boolean;
  /** Age of the oldest message in seconds */
  oldestMessageAgeSeconds?: number;
  /** Number of active conversations */
  activeConversations: number;
  /** Number of conversations in error state */
  errorConversations: number;
  /** Overall health status (Healthy, Warning, Critical, Unknown) */
  status: string;
}

/**
 * Represents a message received from a queue
 */
export interface MessageReceived<T = unknown> {
  /** The queue the message was received from */
  queueName: string;
  /** The conversation handle for this message */
  conversationHandle?: string;
  /** The message type */
  messageType: string;
  /** The deserialized message body */
  body: T;
  /** When the message was received */
  receivedAt: Date;
}

/**
 * Result of sending a message
 */
export interface SendMessageResult {
  /** Whether the send was successful */
  success: boolean;
  /** The conversation handle for the sent message */
  conversationHandle?: string;
  /** Message describing the result */
  message: string;
  /** Error details if the send failed */
  error?: string;
}

/**
 * Request for sending a message
 */
export interface SendMessageRequest {
  /** The target queue name */
  queueName: string;
  /** The message type */
  messageType: string;
  /** The message body (will be serialized to JSON) */
  messageBody: string;
}

File: MessageBus/MessageBus.Client/Models/QueueStatus.cs
namespace MessageBus.Client.Models;

/// <summary>
/// Represents the status of a message queue
/// </summary>
public class QueueStatus
{
    /// <summary>
    /// Name of the queue
    /// </summary>
    public string QueueName { get; set; } = string.Empty;

    /// <summary>
    /// Total number of messages in the queue
    /// </summary>
    public int MessageCount { get; set; }

    /// <summary>
    /// Number of messages ready for processing
    /// </summary>
    public int ReadyCount { get; set; }

    /// <summary>
    /// Whether receive is enabled on the queue
    /// </summary>
    public bool IsReceiveEnabled { get; set; }

    /// <summary>
    /// Whether activation is enabled on the queue
    /// </summary>
    public bool IsActivationEnabled { get; set; }

    /// <summary>
    /// Age of the oldest message in seconds
    /// </summary>
    public int? OldestMessageAgeSeconds { get; set; }

    /// <summary>
    /// Number of active conversations
    /// </summary>
    public int ActiveConversations { get; set; }

    /// <summary>
    /// Number of conversations in error state
    /// </summary>
    public int ErrorConversations { get; set; }

    /// <summary>
    /// Overall health status (Healthy, Warning, Critical, Unknown)
    /// </summary>
    public string Status { get; set; } = string.Empty;
}

/// <summary>
/// Represents a message received from a queue
/// </summary>
public class MessageReceived<T>
{
    /// <summary>
    /// The queue the message was received from
    /// </summary>
    public string QueueName { get; set; } = string.Empty;

    /// <summary>
    /// The conversation handle for this message
    /// </summary>
    public Guid ConversationHandle { get; set; }

    /// <summary>
    /// The message type
    /// </summary>
    public string MessageType { get; set; } = string.Empty;

    /// <summary>
    /// The deserialized message body
    /// </summary>
    public T? Body { get; set; }

    /// <summary>
    /// When the message was received
    /// </summary>
    public DateTime ReceivedAt { get; set; } = DateTime.UtcNow;
}

/// <summary>
/// Result of sending a message
/// </summary>
public class SendMessageResult
{
    /// <summary>
    /// Whether the send was successful
    /// </summary>
    public bool Success { get; set; }

    /// <summary>
    /// The conversation handle for the sent message
    /// </summary>
    public Guid? ConversationHandle { get; set; }

    /// <summary>
    /// Message describing the result
    /// </summary>
    public string Message { get; set; } = string.Empty;

    /// <summary>
    /// Error details if the send failed
    /// </summary>
    public string? Error { get; set; }
}
